src/ctelnet.cpp

```diff
@@ -383,40 +399,67 @@ bool cTelnet::sendData(QString& data)
             outData = outgoingDataEncoder->fromUnicode(data).constData();
         } else {
             // Plain, raw ASCII, we hope!
-            // TODO: Moan if the user is trying to send non-ASCII characters out!
+            for (int i = 0, total = data.size(); i < total; ++i) {
+                if (data.at(i).row() || data.at(i).cell() > 127){
+                    QString errorMsg = tr("[ WARN ] - Invalid characters in outgoing data, one or more characters cannot\n"
+                                          "be encoded into the range that is acceptable for the character\n"
+                                          "encoding that is currently set {\"ASCII\"} for the MUD Server."
+                                          "It may not understand what is sent to it.");
+                    postMessage(errorMsg);
```

Comment:
I think this would do better as a debug window message, not a main window message. Don't need errors and warnings popping up all over, that's why we have the debug window for that!

Reply:
It **is** important though - if you have chosen an encoding that is just plain *wrong* for the character-set that is being used! Although you *can* type the characters in on the command line, and Mudlet *can show* them on the (T)Console it cannot actually **send** them.  It is no good hiding this away in a debug window - it needs the user to do something (change to an Encoding that supports the characters they want to send)...

Reply:
N. B. this error can be produced **directly by the user from the command input widget** as well as from a script/package/module...

Reply:
I do now see that it will be displayed every time there is an encoding problem - which as you say will get spammy.

Let me rework things so that it gets limited to only show in the main console once (so the user will know that there is an issue and what they are typing or sending is not getting through to the Server) and then can only get shown again after the encoding has been changed (*I've done something similar for when an older map format is used and a feature that needs a later one - the `setMapUserData`/`setAreaUserData` lua functions - get used*)...